internal/collector/cloudwatch.go

```diff
@@ -1,57 +1,93 @@
 package collector
 
 import (
+	_ "embed"
+	"html/template"
+	"strings"
+
 	log "github.com/ViaQ/logerr/v2/log/static"
 	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
 	"github.com/openshift/cluster-logging-operator/internal/api/observability"
 	"github.com/openshift/cluster-logging-operator/internal/collector/common"
 	"github.com/openshift/cluster-logging-operator/internal/constants"
 	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/cloudwatch"
-	v1 "k8s.io/api/core/v1"
+	"github.com/openshift/cluster-logging-operator/internal/reconcile"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	CloudwatchCredentialsTemplate = template.Must(template.New("cw credentials").Parse(cloudwatchCredentialsTemplateStr))
+	//go:embed cloudwatch.credentials.tmpl
+	cloudwatchCredentialsTemplateStr string
 )
 
-// Add volumes and env vars if output type is cloudwatch and role is found in the secret
-func addWebIdentityForCloudwatch(collector *v1.Container, forwarderSpec obs.ClusterLogForwarderSpec, secrets observability.Secrets) {
-	if secrets == nil {
-		return
+type CloudwatchWebIdentity struct {
+	Name    string
+	RoleARN string
+	Token   string
```

Comment:
Let's call this "TokenFile" to distiguish it from an actual token. Maybe consider "WebIdentityTokenFile"